includes/Admin/ReviewNotice.php

```diff
@@ -0,0 +1,141 @@
+                                <?php echo __( 'Yes, Absolutely', 'wemail' ); ?>
+                            </button>
+                            <button type="submit" class="button response-button" id="review_reposnse_later" name="review_response_later">
+                                <?php echo __( 'Ask me later', 'wemail' ); ?>
+                            </button>
+                            <button type="submit" class="response-button" id="review_reposnse_no" name="review_response_no">
+                                <u> <?php echo __( 'No, I\'m just taking, not giving', 'wemail' ); ?> </u>
+                            </button>
+                        </form>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <?php
+    }
+
+    /**
+     * Show notice to connect site in weMail
+     *
+     * @return void
+     */
+    public function connect_review_notice() {
+        if ( isset( $_GET['dismiss_wemail_review_notice'] ) && (int) $_GET['dismiss_wemail_review_notice'] === 1 ) {
+            update_option( 'wemail_review_notice', 1 );
+		}
+
+        $installed_time = get_option( 'wemail_installed_time' );
+
+        $this->handle_review_response();
+
+        $this->check_campaign();
+
+		if ( (int) get_option( 'wemail_review_notice' ) !== 1 ) {
+			$this->time_based_review = ( time() - $installed_time ) / 86400 > 7;
+			$this->day_count = (int) ( time() - $installed_time ) / 86400;
+			$this->campaign_count = (int) get_option( 'wemail_sent_campaign_count' );
+			$campaign_based_review = (int) get_option( 'wemail_sent_campaign_count' ) >= 3;
+			if ( $this->time_based_review || $campaign_based_review ) {
+				add_action( 'admin_notices', [ $this, 'connect_review_notice_html' ] );
+			}
+		}
+    }
+
+    public function review_response( $response ) {
+        update_option( 'wemail_response_for_review_notice', $response );
+        update_option( 'wemail_review_notice', 1 );
+    }
+
+    /**
+     * @return void
+     */
+    public function check_campaign() {
+        $campaigns = wemail()->api->campaigns()->get();
+		if ( is_array( $campaigns ) && ! empty( $campaigns['data'] ) && (int) get_option( 'wemail_sent_campaign_count' ) < 3 && get_transient( 'wemail_sent_campaign_count' ) === false ) {
+            $count_campaigns = count( $campaigns['data'] );
+            set_transient( 'wemail_sent_campaign_count', $count_campaigns, 60 * 60 * 2 );
+            update_option( 'wemail_sent_campaign_count', $count_campaigns );
+		}
+    }
+
+    /**
+     * @return void
+     */
+    public function handle_review_response() {
+        $nonce = wp_create_nonce( 'wp_rest' );
+
+        if ( wp_verify_nonce( $nonce, 'wp_rest' ) && isset( $_POST['review_reposnse_yes'] ) ) {
+            $this->review_response( 'yes' );
+            header( 'Location:https://wordpress.org/support/plugin/wemail/reviews/#new-post' );
```

Comment:
WP has a built-in function for redirecting. [wp_safe_redirect](https://developer.wordpress.org/reference/functions/wp_safe_redirect/)